### PR TITLE
Initial docker support

### PR DESCRIPTION
**Pull request by buckley-w-david**
Adding in my initial Dockerfile.  
Ideally needs some more refinement, but this is a good start.

1. Caching the go dependencies will speed up the image build significantly
2. Need a good way to mount in the sqlite database to persist the data, Ideally if we just parameterize the db file location we can mount in a directory and point the server at that directory.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+# Download js dependencies
+COPY makefile .
+COPY web/package.json web/package-lock.json web/
+RUN cd web && npm install && cd .. 
+
+# Download go dependencies and compile
+# TODO: Figure out a way to cache the go dependencies like we do the JS ones to speed up build
+COPY . .
+RUN make npm \
```
Comment (buckley-w-david) [inline Dockerfile]: Don't need this `make npm` anymore with the changes to the makefile, I'll remove it

**Comment by zwzn** [comment]
Do you want to move the project to use modules?

**Comment by zwzn** [comment]
It looks good but I think you will need to merge master to fix the conflicts

**Comment by zwzn** [comment]
It was just the .gitignore I fixed it

**Pull request merged**